jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandlerCollection.java

```diff
@@ -228,7 +227,9 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
         }
         else
         {
-            // This may not work in all circumstances... but then I think it should never be called
+            Handler[] handlers = getHandlers();
```

Comment:
There's not a synchronized(this) on getHandlers, is this safe here?

Reply:
that is fetching a volatile, so there is a memory barrier and we then only act on the returned handlers so we have copy-on-write semantics